packages/clerk-js/src/ui/components/SignUp/SignUpStart.tsx

```diff
@@ -39,7 +39,10 @@ function _SignUpStart(): JSX.Element {
   const { attributes } = userSettings;
   const { setActive } = useClerk();
   const ctx = useSignUpContext();
+  const options = useOptions();
+  const isWithinSignInContext = !!React.useContext(SignInContext);
   const { afterSignUpUrl, signInUrl, unsafeMetadata } = ctx;
+  const isCombinedFlow = (options.experimental?.combinedFlow && !!isWithinSignInContext) || false;
```

Comment:
nit: is `|| false` necessary? What if make it a boolean explicitly?

```suggestion
  const isCombinedFlow = Boolean(options.experimental?.combinedFlow && !!isWithinSignInContext);
```